src/pages/Markets/Recommended.tsx

```diff
@@ -1,18 +1,281 @@
+        .slice(0, 7),
+    [assetIds, selectedChainId],
+  )
+
+  const handleCardClick = useCallback(
+    (assetId: AssetId) => {
+      return history.push(`/assets/${assetId}`)
+    },
+    [history],
+  )
+
+  return (
+    <Grid templateRows={gridTemplateRowsSx} gridTemplateColumns={gridTemplateColumnSx} gap={4}>
+      {filteredAssetIds.map((assetId, index) =>
+        index === 0 ? (
+          <GridItem rowSpan={rowSpanSparklineSx} colSpan={colSpanSparklineSx}>
+            <CardWithSparkline
+              key={assetId}
+              assetId={assetId}
+              onClick={handleCardClick}
+              isLoading={isLoading}
+            />
+          </GridItem>
+        ) : (
+          <GridItem colSpan={colSpanSx}>
+            <AssetCard
+              key={assetId}
+              assetId={assetId}
+              onClick={handleCardClick}
+              isLoading={isLoading}
+            />
+          </GridItem>
+        ),
+      )}
+    </Grid>
+  )
+}
+
+const LpGrid: React.FC<{ assetIds: AssetId[]; selectedChainId?: ChainId; isLoading: boolean }> = ({
+  assetIds,
+  selectedChainId,
+  isLoading,
+}) => {
+  const history = useHistory()
+  const handleCardClick = useCallback(
+    (assetId: AssetId) => {
+      return history.push(`/assets/${assetId}`)
+    },
+    [history],
+  )
+  const { data: portalsAssets } = usePortalsAssetsQuery()
 
+  const filteredAssetIds = useMemo(
+    () =>
+      (selectedChainId
+        ? assetIds.filter(assetId => fromAssetId(assetId).chainId === selectedChainId)
+        : assetIds
+      )
+        // TODO(gomes): remove me when we have real data here for all categories
+        .slice(0, 7),
+    [assetIds, selectedChainId],
+  )
+
+  return (
+    <Grid templateRows={gridTemplateRowsSx} gridTemplateColumns={gridTemplateColumnSx} gap={4}>
+      {filteredAssetIds.map((assetId, index) => {
+        const apy = portalsAssets?.byId[assetId]?.metrics.apy
+        const volume24H = portalsAssets?.byId[assetId]?.metrics.volumeUsd1d
+
+        if (index === 0) {
+          return (
+            <GridItem rowSpan={rowSpanSparklineSx} colSpan={colSpanSparklineSx}>
+              <CardWithSparkline
+                assetId={assetId}
+                onClick={handleCardClick}
+                isLoading={isLoading}
+              />
+            </GridItem>
+          )
+        } else {
+          return (
+            <GridItem colSpan={colSpanSx}>
+              <LpCard
+                assetId={assetId}
+                apy={apy ?? '0'}
+                volume24H={volume24H ?? '0'}
+                onClick={handleCardClick}
+                isLoading={isLoading}
+              />
+            </GridItem>
+          )
+        }
+      })}
+    </Grid>
+  )
+}
+
+const Row: React.FC<RowProps> = ({
+  title,
+  subtitle,
+  children,
+  selectedChainId,
+  setSelectedChainId,
+}) => {
+  const chainIds = Object.values(KnownChainIds)
```

Comment:
[preferably blocking] This could use a useMemo for referential stability

Reply:
I like stability f64fb8ed66